File: VirtualGardens/VirtualGardens.Services/Database/Nalozi.cs
using System;
using System.Collections.Generic;
using VirtualGardens.Services.BaseInterfaces;

namespace VirtualGardens.Services.Database;

public partial class Nalozi:ISoftDeletable
{
    public int NalogId { get; set; }

    public string BrojNaloga { get; set; } = null!;

    public DateTime DatumKreiranja { get; set; } = DateTime.Now;

    public int ZaposlenikId { get; set; }

    public bool Zavrsen { get; set; } = false;

    public virtual ICollection<Narudzbe> Narudzbes { get; set; } = new List<Narudzbe>();

    public virtual Zaposlenici Zaposlenik { get; set; } = null!;
    public bool IsDeleted { get; set; } = false;
    public DateTime? VrijemeBrisanja { get; set; }
}

File: UI/virtualgardens_mobile/lib/models/nalozi.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:virtualgardens_mobile/models/narudzbe.dart';
import 'package:virtualgardens_mobile/models/zaposlenici.dart';

part 'nalozi.g.dart';

@JsonSerializable()
class Nalog {
  int nalogId;
  String brojNaloga;
  DateTime datumKreiranja;
  int zaposlenikId;
  bool zavrsen;
  Zaposlenik? zaposlenik;
  List<Narudzba> narudzbes;

  Nalog({
    required this.nalogId,
    required this.brojNaloga,
    required this.datumKreiranja,
    required this.zaposlenikId,
    required this.zavrsen,
    this.zaposlenik,
    this.narudzbes = const [],
  });

  factory Nalog.fromJson(Map<String, dynamic> json) => _$NalogFromJson(json);

  Map<String, dynamic> toJson() => _$NalogToJson(this);
}
